frontend/catalyst/pennylane_extensions.py

```diff
@@ -115,19 +116,37 @@ class QFunc:
             the valid gate set for the quantum function
     """
 
-    def __init__(self, fn, device):
+    def __init__(self, fn, device):  # pragma: nocover
         self.func = fn
         self.device = device
         update_wrapper(self, fn)
 
+    @staticmethod
+    def _add_toml_file(device):
+        """Temporary function. This function adds the config field to devices.
+        TODO: Remove this function when `qml.Device`s are guaranteed to have their own
+        config file field."""
+        device_lpath = pathlib.Path(get_lib_path("runtime", "RUNTIME_LIB_DIR"))
+        name = device.name
+        if isinstance(device, qml.Device):
+            name = device.short_name
+
+        # The toml files name convention we follow is to replace
+        # the dots with underscores in the device short name.
+        toml_file_name = name.replace(".", "_") + ".toml"
```

Comment:
If this is required, should it be mentioned in the docs? Or will implementers following the docs not need this because they already provide a config file field?

Reply:
It is not necessary. This is just a temporary solution before we catch up to pennylane-lightning / latest and before other devices have the configuration field set. The name can be completely arbitrary, but the pennylane-lightning team wanted the name `lightning_qubit.toml`. So I just made them similar.